util/sponsor/auth.go

```diff
@@ -16,19 +16,30 @@ var (
 	ExpiresAt      time.Time
 )
 
-const unavailable = "sponsorship unavailable"
+const (
+	unavailable = "sponsorship unavailable"
+	victron     = "victron-device"
+)
 
 func IsAuthorized() bool {
 	return len(Subject) > 0
 }
 
 func IsAuthorizedForApi() bool {
-	return IsAuthorized() && Subject != unavailable
+	return IsAuthorized() && Token != ""
 }
 
 // check and set sponsorship token
 func ConfigureSponsorship(token string) error {
 	if token == "" {
+		if valid, err := isVictron(); valid && err == nil {
```

Comment:
Die Fehlerbehandlung überblicke ich nicht. Geht das einfacher?

Reply:
Ja, verstehe was du meinst. Im Gegensatz zu anderen Stellen wird Error nicht einfach hochgereicht sondert meint hier explizit `sponsorship check error`.

Wir haben hier ja mehrere Fälle:
- Hardware liefert nicht alle Werte: `valid=false, error=nil`
- Werte vorhanden, Sponsor-Check durchgeführt: `valid=[true|false], error=nil`
- Werte vorhanden, Sponsor-Check nicht möglich: `valid=false, error=<error>`

Diese Fälle und das Schreiben von `Subject` wollte ich gerne `ConfigureSponsorship` sichtbar behalten. Technisch könnten wir das auch alles in `isVictron` machen und danach einfach nur prüfen (sideffect) ob `Subject` einen Wert bekommen hat. Fändest du das schöner? Kapselt die Victron Logik stärker, arbeitet aber mit Globals anstatt mir Rückgaben. Das spart uns den Umweg über `valid`/`error`.

Reply:
@andig schau noch mal drüber. Die Methode liefert nun direkt das subject zurück. Heißt die Indirektion über valid/error ist damit raus.